### PR TITLE
Call out rate limiting for public buckets

### PR DESCRIPTION
**Pull request by mnewswanger-cloudflare**
Add notes around public bucket usage and rate limiting

### CODE REVIEW

**Review by bthwaites — APPROVED** [review]

File: content/r2/data-access/public-buckets.md [hunk 1]
```diff
@@ -99,6 +99,14 @@ The domain is no longer connected to your bucket and will no longer appear in th
 Enabling managed public access for your bucket will make the content of your bucket available to the Internet through a `r2.dev` Cloudflare subdomain.
 To enable public access for your buckets:
 
+{{<Aside type="note">}}
+
+Public access through `r2.dev` subdomains are rate limited and should only be used for development purposes.
+
+To enable access management, Cache and bot management features, you must set up a custom domain when enabling public access to your bucket.
```
Comment (harshal317) [inline content/r2/data-access/public-buckets.md:106]: Should this maybe be instead of enabling public access to your bucket

**Review by harshal317 — APPROVED** [review]

File: content/r2/data-access/public-buckets.md (hunk 1 above)
Comment (Frederik-Baetens) [inline content/r2/data-access/public-buckets.md:106]: How about something like:

```suggestion
For production workloads requiring public access to your bucket, you must set up a custom domain. This also enables the use of Cache, bot management and access management features.
```

That makes it clearer that custom domains are intended for production workloads and that the difference isn't just those extra features.

Comment (mnewswanger-cloudflare) [reply content/r2/data-access/public-buckets.md:106]: The latter bit there is just moving up from lower in the doc (no changes here). I actually prefer the phrasing calling out the functionality first followed by what action users need to take.

Comment (Frederik-Baetens) [reply content/r2/data-access/public-buckets.md:106]: Yeah, but I think we should explicitly communicate that custom domains are intended for production workloads/don't have rate limits. I already saw confusion about this in the discord.

Comment (mnewswanger-cloudflare) [reply content/r2/data-access/public-buckets.md:106]: That copy's unchanged from earlier (just moved up so the info's above the steps to enable). I think the phrasing gets a little confusing with the phrasing of `Managed Pubic Access` vs more generic "public access" so I'd lean toward just doing nothing as it's already been reviewed.

Comment (mnewswanger-cloudflare) [reply content/r2/data-access/public-buckets.md:106]: I'll merge this one as-is for now; if we continue to see noise and confusion related to it, we can revisit and easily update later.

**Pull request merged**